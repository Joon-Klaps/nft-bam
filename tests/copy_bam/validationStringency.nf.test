nextflow_process {

    name "Test Process COPY_BAM"
    script "./main.nf"
    process "COPY_BAM"

    tag "stringency"

    test("don't fail when set to lenient") {
        when {
            process {
                """
                input[0] = Channel.of(
                    file("${baseDir}/tests/error_hg002.bam", checkIfExists:true)
                )
                """
            }
        }

        then {
            assert snapshot(process.out.bam.collect { 
                bam(it, stringency: "lenient").getReadsMD5()
            }).match("lenient")
        }       
    }

    test("don't fail when set to silent") {
        when {
            process {
                """
                input[0] = Channel.of(
                    file("${baseDir}/tests/error_hg002.bam", checkIfExists:true)
                )
                """
            }
        }

        then {
            assert snapshot(process.out.bam.collect { 
                bam(it, stringency: "silent").getReadsMD5()
            }).match("silent")
        }       
    }

    test("fail on unknown stringency") {
        when {
            process {
                """
                input[0] = Channel.of(
                    file("${baseDir}/tests/error_hg002.bam", checkIfExists:true)
                )
                """
            }
        }

        then {
            try {
                bam(process.out.bam.get(0), stringency: "unknown").getReads()
            } catch( IllegalArgumentException e ) {
                assert snapshot(
                    e.getMessage()
                ).match("unknown")
            }
        }       
    }
}
