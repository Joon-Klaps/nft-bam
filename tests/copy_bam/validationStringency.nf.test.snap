{
    "silent": {
        "content": [
            [
                "bb6e2fb0828114f50f9f0f7c397ec5e4"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-07-01T16:58:15.601665934"
    },
    "lenient": {
        "content": [
            [
                "bb6e2fb0828114f50f9f0f7c397ec5e4"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-07-01T16:57:59.925013954"
    },
    "unknown": {
        "content": [
            "The validation stringency should be one of these: 'STRICT', 'LENIENT', 'SILENT'. Found: UNKNOWN"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-07-01T16:56:56.176765731"
    }
}